apps/cms.c

```diff
@@ -1103,7 +1122,11 @@ int cms_main(int argc, char **argv)
             key = NULL;
         }
         /* If not streaming or resigning finalize structure */
-        if ((operation == SMIME_SIGN) && !(flags & CMS_STREAM)) {
+        if (operation == SMIME_SIGN && digestbin != NULL && !(flags & CMS_STREAM)) {
+            /* Use pre-computed digest instead of content */
+            if (!CMS_final_digest(cms, digestbin, digestlen, NULL, flags))
+                goto end;
+        } else if ((operation == SMIME_SIGN) && !(flags & CMS_STREAM)) {
```

Comment:
Please drop the extra parentheses around the first condition and use `(flags & CMS_STREAM) == 0`